src/test/ui/span/issue-71363.rs

```diff
@@ -1,6 +1,7 @@
-// compile-flags: -Z simulate-remapped-rust-src-base=/rustc/xyz -Z ui-testing=no
-// only-x86_64-unknown-linux-gnu
-//---^ Limiting target as the above unstable flags don't play well on some environment.
+// FIXME(#97682):
+// The `-Z simulate-remapped-rust-src-base=/rustc/xyz -Z ui-testing=no` flags
+// don't work well and make UI test fail on some env.
+// Once it starts to work fine, we could re-enable them here.
```

Comment:
This test isn't disabled, right? I think we probably want to mark it as ignore for now, so that we can more easily tell that the test is not actually testing the right behavior and needs fixing.

Unless these flags are added for unrelated reasons? I don't see an obvious mention of them in the original issue (#71363)....

Reply:
Yeah, this PR disabled the flags, but didn't disable the test itself.


> This test isn't disabled, right? I think we probably want to mark it as ignore for now, so that we can more easily tell that the test is not actually testing the right behavior and needs fixing.
> 
> Unless these flags are added for unrelated reasons? I don't see an obvious mention of them in the original issue (#71363)....

I just copy-pasted them from the previous PR so I cannot explain it correctly, I guess this comment has some context?: https://github.com/rust-lang/rust/issues/71363#issuecomment-616780419

Reply:
Added `// ignore-test`.

Reply:
Removing the flag would indeed make the test incorrect.

This test ensures that when a file (in this case `library/std/src/error.rs`) is not included in the error message it will not affect the indentation of the error message itself (due to the line number length).

The test tried to rely on `-Z simulate-remapped-rust-src-base` to exclude the `library/std/src/error.rs` file from the error message, but that flag is buggy. The flag simulated `remap-debuginfo = true` only partially, and crucially didn't simulate mapping a remapped path to a local path when possible. Because of this, when `remap-debuginfo = false` the file wasn't mapped to a local path (excluding it from the error), while when `remap-debuginfo = true` it was mapped to a local file, causing the test to fail.

I opened https://github.com/rust-lang/rust/pull/97786 to fix the behavior of that flag, but it also makes the test incorrect (as it always maps the remapped path to a local file).

If we'll want to fix this test we need to either find a path that will not be included in the error message regardless of `remap-debuginfo`, or add a new `-Z` flag [to skip this piece of code](https://github.com/rust-lang/rust/blob/6609c6734de4df43e24d7672f8ae8786ecc8047e/compiler/rustc_metadata/src/rmeta/decoder.rs#L1479-L1556).

Reply:
So, does your PR fix the different behavior of this test described in #97682? If so, then we could close this and reopen #71363. This PR's motivation was to make UI test pass on @[japaric](https://github.com/japaric)'s env and ignoring would be sufficient for that purpose.

Reply:
Yes. I'm also working on a PR to try and add a `-Z` flag to make the test work as expected.